MinGW build, tests fail and crash.

Some tests in the `test.c` (cbbc4a6) fail when building with MinGW clang 10.0.0 and GCC 10.1.0 on Windows 10.

Following output is produces:
```
test.c(129): TEST FAILED: "2.118018798719000895" != "2.118018798719000900"
test.c(135): TEST FAILED: "2.918018798719000717" != "2.918018798719000700"
test.c(141): TEST FAILED: "2.518018798719000806" != "2.518018798719000800"
test.c(147): TEST FAILED: "-2.118018798719000895" != "-2.118018798719000900"
test.c(153): TEST FAILED: "-2.918018798719000717" != "-2.918018798719000700"
test.c(159): TEST FAILED: "2.518018798719000806" != "2.518018798719000800"
test.c(193): TEST FAILED: "+0.99990000000000001101341240428155288100242614746094" != "+0.99990000000000001000000000000000000000000000000000"
test.c(201): TEST FAILED: "0.00000000000000000005421010862427522170037264004349708557128906250000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000" != "0.00000000000000000005421010862427522200000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
test.c(206): TEST FAILED: "0.0000000000000000000542101086242752217003726400434970855712890625000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000" != "0.0000000000000000000542101086242752220000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
```

Building with `-Wextra` give following warnings:
```
In file included from test.c:9:
../r128.h:801: warning: ignoring '#pragma code_seg ' [-Wunknown-pragmas]
  801 | #pragma code_seg(".text")
      |
../r128.h:802:1: warning: 'allocate' attribute directive ignored [-Wattributes]
  802 | __declspec(allocate(".text")) static const unsigned char r128__udiv128Code[] = {
      | ^~~~~~~~~~
../r128.h: In function 'r128__format':
../r128.h:1191:25: warning: comparison of integer expressions of different signedness: 'int' and 'long long unsigned int' [-Wsign-compare]
 1191 |    } else if (precision > sizeof(buf) - 21) {
      |                         ^
```

GCC 9.3.0 on Linux and MSVC 2019 build (cl 19.25.28614) is fine (all test pass).

--- a/r128.h
+++ b/r128.h
@@ -795,7 +795,7 @@ static void r128__umul128(R128 *dst, R128_U64 a, R128_U64 b)
 }
 
 // 128/64->64
-#if defined(_M_X64) && (_MSC_VER < 1920) && !defined(R128_STDC_ONLY)
+#if defined(_M_X64) && (_MSC_VER < 1920) && !defined(R128_STDC_ONLY) && !defined(__MINGW32__)
 // MSVC x64 provides neither inline assembly nor (pre-2019) a div intrinsic, so we do fake
 // "inline assembly" to avoid long division or outline assembly.
 #pragma code_seg(".text")
@@ -810,7 +810,7 @@ static const r128__udiv128Proc r128__udiv128 = (r128__udiv128Proc)(void*)r128__u
 #else
 static R128_U64 r128__udiv128(R128_U64 nlo, R128_U64 nhi, R128_U64 d, R128_U64 *rem)
 {
-#if defined(_M_X64) && !defined(R128_STDC_ONLY)
+#if defined(_M_X64) && !defined(R128_STDC_ONLY) && !defined(__MINGW32__)
    return _udiv128(nhi, nlo, d, rem);
 #elif defined(__x86_64__) && !defined(R128_STDC_ONLY)
    R128_U64 q, r;

--- a/test/test.c
+++ b/test/test.c
@@ -6,6 +6,11 @@
 #ifndef NDEBUG
 #  define R128_DEBUG_VIS
 #endif
+
+#ifdef __MINGW32__
+#  define __USE_MINGW_ANSI_STDIO 1 // force standard sprintf precision
+#endif
+
 #include "../r128.h"
 
 #include <math.h>

